packages/js/product-editor/src/components/header/header.tsx

```diff
@@ -1,14 +1,38 @@
 /**
  * External dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { createElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Product } from '@woocommerce/data';
 
 export type HeaderProps = {
+	product: Product;
 	title: string;
 };
 
-export function Header( { title }: HeaderProps ) {
+export function Header( { product, title }: HeaderProps ) {
+	const { isProductLocked, isSaving } = useSelect(
+		( select ) => {
+			const { isPostSavingLocked, isSavingPost } =
+				select( 'core/editor' );
+			return {
+				isProductLocked: isPostSavingLocked(),
+				isSaving: isSavingPost(),
+			};
+		},
+		[ product.id ]
```

Comment:
We can remove this now right? and in turn also remove the `product` prop from this component?
I believe the `core/editor` should re-trigger this `useSelect` if the active post gets updated, so listening to the `id` won't be necessary.

Reply:
Ohh! right! Thanks

Reply:
Done -> [193d285](https://github.com/woocommerce/woocommerce/pull/37152/commits/193d285a608f1d347b1d3a2bc2c006abbdab3609) Thanks. Good catch.